### DISCUSSION TITLE
ListBox and WhenActivated

### DISCUSSION THREAD

**Discussion by ctrl-alt-d** [AvaloniaUI/Avalonia#6420] [Q&A] [answered]
I use a `ListBox` to show items as "rows", each item has a `UserControl` with a `ViewModel`. 

This is how I'm registering the handler to open the edit dialog on each row:

```c#
        public CentreRowUserCtrl()
        {
            InitializeComponent();
            this.WhenActivated(d => d(ViewModel!.ShowDialog.RegisterHandler(UpdateShowDialogAsync)));
        }
```

I guess they are some issues with virtualization and only the firsts `ViewModel!.ShowDialog`'s are bound. Also, the whole row remains on gray when I click the menu. Some other rows also become gray I don't know why, but I suspect all is related with virtualization:

![listbox](https://user-images.githubusercontent.com/3105983/129449615-45e7e1d4-d64c-4ffa-973a-c5c36dd772a2.gif)

How can I fix it? All run fine if I use `ItemsControl` instead `ListBox`, but, they are serious performance issues with `ItemsControl`:  https://github.com/AvaloniaUI/Avalonia/discussions/6417

**Comment by ctrl-alt-d** [comment, marked as answer]
Related: https://github.com/AvaloniaUI/Avalonia/discussions/6427

**Reply by maxkatz6** [reply]
Was your question answered with https://github.com/AvaloniaUI/Avalonia/discussions/6427 ?

**Reply by ctrl-alt-d** [reply]
#6427 closes this issue. Using 

```c#
this.WhenActivated(disposables => {
    this.WhenAnyValue(x => x.ViewModel)
        .Subscribe(vm => { /* Here you can access the view model safely. */ })
        .DisposeWith(disposables);
});
```

Was the solution.
